spec/openapi.yaml

```diff
@@ -656,6 +662,185 @@ servers:
+          required: true
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: 2019-12-12T00:00:00.000+11:00
+        - name: end_date
+          in: query
+          description: End date time for report window. By default, the timezone for this parameter will be taken from the account settings for the account associated with the credentials used to make the request, or the account included in the Account parameter. This can be overridden using the timezone parameter per request. The date must be in ISO8601 format.
+          required: true
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: 2019-12-14T00:00:00.000+11:00
+        - name: direction
+          in: query
+          description: Start date time for report window. By default, the timezone for this parameter will be taken from the account settings for the account associated with the credentials used to make the request, or the account included in the Account parameter. This can be overridden using the timezone parameter per request. The date must be in ISO8601 format.
+          style: form
+          explode: true
+          schema:
+            type: object
+        - name: source
+          in: query
+          description: Filter results by source address.
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: +61555555555
+        - name: destination
+          in: query
+          description: Filter results by destination address.
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: +61555555555
+        - name: destinations
+          in: query
+          description: Specify a list of destination numbers to filter the report by
+          style: form
+          explode: true
+          schema:
+            type: array
+            items:
+              type: string
+        - name: metadata_key
+          in: query
+          description: Filter results for messages that include a metadata key.
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: broadcastId
+        - name: metadata_value
+          in: query
+          description: Filter results for messages that include a metadata key containing this value. If this parameter is provided, the metadata_key parameter must also be provided.
+          style: form
+          explode: true
+          schema:
+            type: string
+            example: ABC
+        - name: metadata_values
+          in: query
+          description: Specify a list of metadata values to filter the report by
```

Comment:
We should include the requirement for `metadata_key` here too as you have for `metadata_value`:
> If this parameter is provided, the metadata_key parameter must also be provided.

Reply:
So make the `metadata_key` parameter have `required: true`? Wouldn't it only be required in the case of having `metadata_value` present? I'm not sure OpenAPI can support conditional requirements or not?

Reply:
Sorry no, not a hard requirement for every request to warrant `required: true` but just as you have done with `metadata_value`, `metadata_values` should include the dependency on `metadata_key` in the description.
- `metadata_key` can be used independently of `metadata_value` and `metadata_values`
- `metadata_value` requires `metadata_key` for expected behaviour
- `metadata_values` requires `metadata_key` for expected behaviour

Thinking something like:
> Specify a list of metadata values to filter for messages including a filtered metadata value. If this parameter is provided, the metadata_key parameter must also be provided.